Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1345,9 +1345,9 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
         // - Items placed at the first implicit line in the masonry axis.
         // - Items that have a specified definite placement in the grid axis.
         // - Items that span all grid axis tracks.
-        if (m_renderGrid->areMasonryRows() || m_renderGrid->areMasonryColumns()) {
+        if (m_renderGrid->isMasonry()) {
             bool skipTrackSizing = true;
-            auto gridAxisDirection = m_renderGrid->areMasonryRows() ? ForColumns : ForRows;
+            auto gridAxisDirection = m_direction;
```

Comment:
`ASSERT(!isDirectionInMasonryDirection())` ?

Just for extra clarity when reading that m_direction is never the masonry direction.